pyproject.toml

```diff
@@ -31,7 +31,9 @@ dependencies = [
     "aws-cdk.aws-lambda-python-alpha",
```

Comment:
This isn't part of your changeset, but why is the pyproject.toml set up like this? The documentation has it so that your dependencies are under their own header: `[tool.poetry.dependencies]`. This gives more flexibility for different groupings or optional dependencies. Also, I'm just in general kind of confused about the formatting here, it doesn't seem to align with what the docs say. 

https://python-poetry.org/docs/master/pyproject/#dependencies-and-dependency-groups

Reply:
I'm happy to change the pyproject format and make the changes I suggested btw, especially since this isn't really part of your change set so it shouldn't block the merge. I just want to make sure this format isn't due to some codespace rule or some formatting recommendations that I am unaware of.

Reply:
This is one of my biggest issues with the python packaging trying to move over to this new format now. Each tool/backend will define various subsections and so it is confusing to change back and forth for requirements. This is currently correct for the setuptools backend, but looks like you say it won't work with Poetry?

https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html

Everyone wants to define their own standard 😂

Reply:
I thought this was a poetry pyproject.toml, I didn't realize it was setuptools! Poetry does use a slightly different format, but it won't be hard to transition over once. It's sort of an ["in addition" thing](https://bbc.github.io/cloudfit-public-docs/packaging/this_way_up.html#pyprojecttoml), rather than replacing it. The [project] section can remain the same (I think?), it looks like poetry operates on top of pyproject.toml, not replacing it. 

I hate python dependency management 🙃

Reply:
I would like to move to poetry because it's flexible and powerful enough to completely replace pip, venv, conda, all those managers, while generally [being better than them](https://blogs.sap.com/2022/05/08/why-you-should-use-poetry-instead-of-pip-or-conda-for-python-projects/), particularly at dependency management. 

Dependency management in pip is awful, and in poetry it's still awful but at least you can fix the problems for everyone else? It's not a fix-all but it at least collapses all the dependency management, venv, installation, etc into one thing.